framework/src/positions/Positions.C

```diff
@@ -92,10 +94,18 @@ Positions::getNearestPosition(const Point & target, const bool initial) const
     // Check that no two positions are equidistant to the target
     else if (MooseUtils::absoluteFuzzyEqual((_positions[i] - target).norm_sq(),
                                             nearest_distance_sq))
-      mooseWarning(
-          "Search for nearest position found several matches: " + Moose::stringify(_positions[i]) +
-              " and " + Moose::stringify(_positions[nearest_index]),
-          " at a distance of " + std::to_string(std::sqrt(nearest_distance_sq)));
+      conflict_index = std::make_pair(cast_int<int>(i), cast_int<int>(nearest_index));
+  }
+
+  // If the closest positions was a conflict, report a warning
+  if (cast_int<int>(nearest_index) == conflict_index.first ||
+      cast_int<int>(nearest_index) == conflict_index.second)
```

Comment:
```suggestion
  if (cast_int<int>(nearest_index) == conflict_index.second)
```
It should only be possible to match the second right?

Reply:
sure